src/Containers/Microsoft.NET.Build.Containers/Registry/DefaultRegistryAPI.cs

```diff
@@ -26,9 +28,24 @@ internal DefaultRegistryAPI(Uri baseUri, ILogger logger)
 
     public IManifestOperations Manifest { get; }
 
-    private static HttpClient CreateClient(Uri baseUri, bool isAmazonECRRegistry = false)
+    private static HttpClient CreateClient(string registryName, Uri baseUri, bool isAmazonECRRegistry = false)
     {
-        HttpMessageHandler clientHandler = new AuthHandshakeMessageHandler(new SocketsHttpHandler() { PooledConnectionLifetime = TimeSpan.FromMilliseconds(10 /* total guess */) });
+        var innerHandler = new SocketsHttpHandler()
+        {
+            PooledConnectionLifetime = TimeSpan.FromMilliseconds(10 /* total guess */)
+        };
+
+        // Ignore certificate for https localhost repository.
+        if (baseUri.Host == "localhost" && baseUri.Scheme == "https")
```

Comment:
The new test uses https for localhost. This ignores the certificate errors.

Reply:
Interesting - I thought that [trusting the cert](https://github.com/dotnet/sdk/pull/33753/files#diff-67f5d802410fe8aed729fc7bb9e82da95776c54b6995da4dd76e3edcea5ba837R61-R62) would have satisfied any cert validity checks here.

Reply:
`dev-certs --trust` has never worked on Linux.
I'm not sure it does what you expect on Windows.

I'll remove those lines from the test.

Reply:
I thought the ASP.NET team had been working on revamping trust support on Linux? [This issue](https://github.com/dotnet/aspnetcore/issues/32842) makes me believe that they'd done it?

Reply:
Snippet from dev-cert's `UnixCertificateManager.cs`:

```cs
    protected override void TrustCertificateCore(X509Certificate2 certificate) =>
        throw new InvalidOperationException("Trusting the certificate is not supported on linux");
```

Reply:
ah - and reading that issue a bit further, the missing step is copying the generated cert into the ca-certificates location an updating the cert store:

```
sudo cp generated-cert.crt /usr/local/share/ca-certificates
sudo update-ca-certificates
```

that would be _interesting_ to orchestrate from a test

Reply:
I have some history with the `dev-certs` command and trying to make it work on Linux. Besides adding the certs to the system store (which is different per distro), another challenge is that some Linux tools (which may include .NET HttpClient (through OpenSSL)) don't like self-signed certs for https servers. They require a CA signed cert.

Anyway, the behavior added here seems to match what `docker` does since it simply accepts the cert as-is for `localhost`. `podman` requires the `--tls-verify=false` for this.